src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROWING_COEFFICIENT = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (indexIsInvalid(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index passed");
+        }
+        growIfArrayFull();
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        set(value, index);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (indexIsInvalid(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index passed");
+        }
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (indexIsInvalid(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index passed");
+        }
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (indexIsInvalid(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index passed");
+        }
+        T object = get(index);
+        System.arraycopy(elementData, index + 1, elementData, index, size - index - 1);
+        size--;
+        return object;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (elementsAreEquals(element, elementData[i])) {
+                remove(i);
+                return element;
+            }
+        }
+        throw new NoSuchElementException("No such element present");
```

Comment:
Also, add more information.
```suggestion
        throw new NoSuchElementException("No such element present: " + element);
```